src/main/groovy/org/jetbrains/intellij/tasks/RunPluginVerifierTask.groovy

```diff
@@ -0,0 +1,354 @@
+
+    @Input
+    @Optional
+    String getTeamCity() {
+        return teamCity
+    }
+
+    void setTeamCity(String teamCity) {
+        this.teamCity = teamCity
+    }
+
+    @Input
+    @Optional
+    String getTcGrouping() {
+        return tcGrouping
+    }
+
+    void setTcGrouping(String tcGrouping) {
+        this.tcGrouping = tcGrouping
+    }
+
+    @Input
+    @Optional
+    String getExcludedPluginsFile() {
+        return excludedPluginsFile
+    }
+
+    void setExcludedPluginsFile(String excludedPluginsFile) {
+        this.excludedPluginsFile = excludedPluginsFile
+    }
+
+    @Input
+    @Optional
+    String getDumpBrokenPluginList() {
+        return dumpBrokenPluginList
+    }
+
+    void setDumpBrokenPluginList(String dumpBrokenPluginList) {
+        this.dumpBrokenPluginList = dumpBrokenPluginList
+    }
+
+    @Input
+    @Optional
+    String getPluginsToCheckFile() {
+        return pluginsToCheckFile
+    }
+
+    void setPluginsToCheckFile(String pluginsToCheckFile) {
+        this.pluginsToCheckFile = pluginsToCheckFile
+    }
+
+    @Input
+    @Optional
+    String getSubsystemsToCheck() {
+        return subsystemsToCheck
+    }
+
+    void setSubsystemsToCheck(String subsystemsToCheck) {
+        this.subsystemsToCheck = subsystemsToCheck
+    }
+
+    @TaskAction
+    void runPluginVerifier() {
+        def extension = project.extensions.findByType(IntelliJPluginExtension)
+        def pluginFileName = "${extension.pluginName}-${project.version}"
+
+        if (!project.file("${project.buildDir}/distributions/${pluginFileName}.zip").exists()) {
+            throw new IllegalStateException("Plugin file $pluginFileName does not exist.")
+        }
+
+        if (ides.isEmpty()) {
+            ides.add("${extension.type}-${extension.version}")
+        }
+
+        def resolver = new IdeaDependencyManager(extension.intellijRepo ?: IntelliJPlugin.DEFAULT_INTELLIJ_REPO)
+        def verifierPath = getVerifierPath()
+
+        def verifierArgs = ["check-plugin"]
+        verifierArgs += getOptions()
+        verifierArgs += ["${project.buildDir}/distributions/${pluginFileName}.zip"]
+        verifierArgs += ides.collect {
+            def (String type, String version) = it.split("-")
+            def dependency = resolver.resolveRemote(project, version, type, false)
+            return dependency.classes.absolutePath
+        }
+
+        new ByteArrayOutputStream().withStream { os ->
+            project.javaexec {
+                classpath = project.files(verifierPath)
+                main = "com.jetbrains.pluginverifier.PluginVerifierMain"
+                args = verifierArgs
+                standardOutput = os
+            }
+
+            def output = os.toString()
+            println output
+            for (FailureLevel level : FailureLevel.values()) {
+                if (failureLevel.contains(level) && output.contains(level.testValue)) {
+                    throw new GradleException(level.toString())
```

Comment:
Do we really need to fail the entire build here?

Reply:
It's supposed to fail only if the user wants it failing by setting the _failureLevel_ property. I find it useful in the case of CI to fail the job.